lib/compat/time.c

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2016 Balabit
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include "compat/time.h"
+
+#ifdef __MACH__
+#include <mach/clock.h>
+#include <mach/mach.h>
+#endif
+
+struct timespec
+get_monotonic_time(void)
+{
+  struct timespec timestamp;
+#ifdef __MACH__
+  clock_serv_t clock_server;
+  mach_timespec_t mach_timestamp;
+  host_get_clock_service(mach_host_self(), SYSTEM_CLOCK, &clock_server);
+  clock_get_time(clock_server, &mach_timestamp);
+  mach_port_deallocate(mach_task_self(), clock_server);
+  timestamp.tv_sec = mach_timestamp.tv_sec;
+  timestamp.tv_nsec = mach_timestamp.tv_nsec;
+#else
+  clock_gettime(CLOCK_MONOTONIC, &timestamp);
+#endif
```

Comment:
If you intend to move this function into combat then please use and adhere to the POSIX prototype (e.g. implement clock_gettime() and not a wrapper). This is the convention we use in other parts of contrib.

If you want to create a wrapper then do that in lib and not compat.

Thanms